src/python_inspector/utils.py

```diff
@@ -54,6 +54,20 @@ def write_output_in_file(output, location):
     return output
 
 
+def write_resolved_packages(package_list, requirements_file):
+    """
+    Write the resolved package names and versions into ``requirements_file_path``
+    """
+    dependencies = package_list[0]["package_data"][0]["dependencies"]
```

Comment:
what if `package_list` is empty? and `package_data` does not exist there

Reply:
I have implemented the checks for empty or corrupt `package_list`.